frontend/src/components/DropdownMenu.js

```diff
@@ -0,0 +1,155 @@
+import React, { useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import './DropdownMenu.css';
+import triangleDown from '../images/triange_down.png';
+
+export default function DropdownMenu({
+  buttonText,
+  buttonAriaLabel,
+  styleAsSelect,
+  canBlur,
+  children,
+  disabled,
+  applyButtonText,
+  applyButtonAria,
+  onApply,
+  className,
+  menuName,
+  onCancel,
+  showCancel,
+  cancelAriaLabel,
+  forwardedRef,
+}) {
+  const [menuIsOpen, setMenuIsOpen] = useState(false);
+  const menuContents = useRef();
+
+  /**
+   * Close the menu on escape key
+   * @param {Event} e
+   */
+  const onKeyDown = (e) => {
+    if (e.keyCode === 27) {
+      setMenuIsOpen(false);
+    }
+  };
+
+  /**
+   * Close the menu on blur, with some extenuating circumstance
+   *
+   * @param {Event} e
+   * @returns void
+   */
+  const onBlur = (e) => {
+    // if we're within the same menu, do nothing
+    if (e.relatedTarget && menuContents.current.contains(e.relatedTarget)) {
```

Comment:
Seems like our usage of Node.contains should be fully supported by ie11
https://developer.mozilla.org/en-US/docs/Web/API/Node/contains

As far as I understand, we will only be calling it on instances of HTMLElement, not a text node or anything like that